$OpenBSD: patch-src_xitk_playlist_c,v 1.1 2019/12/28 10:08:17 ajacoutot Exp $

Fix some crashes.

Eg, when starting a playlist while already playing something else.

Index: src/xitk/playlist.c
--- src/xitk/playlist.c.orig
+++ src/xitk/playlist.c
@@ -260,7 +260,7 @@ void playlist_delete_entry(int j) {
         enable_playback_controls (gui->panel, 0);
 
       if(xine_get_status(gui->stream) != XINE_STATUS_STOP)
-	gui_stop(NULL, NULL);
+	gui_stop(NULL, gui);
 
       gui_set_current_mmk(NULL);
       xitk_inputtext_change_text(playlist->winput, NULL);
@@ -294,7 +294,7 @@ void playlist_delete_all(xitk_widget_t *w, void *data)
   playlist_update_playlist();
   
   if(xine_get_status(gui->stream) != XINE_STATUS_STOP)
-    gui_stop(NULL, NULL);
+    gui_stop(NULL, gui);
   
   if(playlist && playlist->winput)
     xitk_inputtext_change_text(playlist->winput, NULL);
@@ -375,7 +375,7 @@ static void _playlist_load_callback(filebrowser_t *fb)
     playlist_update_playlist();
 
     if((xine_get_status(gui->stream) == XINE_STATUS_PLAY))
-      gui_stop(NULL, NULL);
+      gui_stop(NULL, gui);
 
     if ((!is_playback_widgets_enabled (gui->panel)) && gui->playlist.num)
       enable_playback_controls (gui->panel, 1);
@@ -829,7 +829,7 @@ void playlist_scan_input(xitk_widget_t *w, void *ip) {
      */
     if (gui->smart_mode) {
       if (xine_get_status (gui->stream) == XINE_STATUS_PLAY)
-        gui_stop (NULL, NULL);
+        gui_stop (NULL, gui);
       gui_play (NULL, gui);
     }
     
